enhancements/sdk-migrate-bundle.md

```diff
@@ -0,0 +1,327 @@
+bundles which is the preferred OLM packaging format.
+
+### Goals
+
+- Provide a subcommand to migrate one packagemanifest to a set of bundles.
+
+### Non-Goals
+
+- Solve the world?
+- There will be no migration from bundles to packagemanifests
+
+## Proposal
+
+### User Stories [optional]
+
+#### Story 1
+
+As an operator developer, I would like to migrate my operator which uses
+the package manifest format to the new bundle format.
+
+#### Story 2
+
+As an catalog admin for my operators, I would like to be able to migrate a
+collection of packagemanifests to bundles.
+
+### Implementation Details/Notes/Constraints [optional]
+
+The Operator SDK currently supports generating & validating
+packagemanifests & bundles. This enhancement will add a new command, `migrate
+bundle` to the CLI. This new command will migrate an existing packagemanifest to
+a bundle.
+
+More information about the different format can be found at the URLs below:
+
+- Operator bundles: https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md
+- Package manifests format: https://github.com/operator-framework/operator-registry/tree/v1.5.3/#manifest-format
+
+The Operator SDK will traverse the given packagemanifest directory and for each
+of the versions create a new bundle.
+
+The Operator SDK will make use of the API available from the Operator
+Registry's `opm` command. The command is pretty well segmented that we could
+easily reuse the `GenerateFunc` function to create the `bundle` as a
+directory. There is also support for building the bundle as an image which
+we could do using the `BuildBundleImage` function.
+
+#### Generating bundle to disk
+
+The `GenerateFunc` takes in the following arguments:
+
+- directory string - local directory where the manifests and metadata are
+  located
+- outputDir string - where manifests and metadata are copied
+- packageName string - the name of the package that bundle image belongs to
+- channels string - the list of channels that bundle image belongs to
+- channelDefault string - the default channel for the bundle image
+- overwrite bool - flag to enable overwriting annotations.yaml
+
+In order to use this function, we will obtain the required parameters from the
+listed *source*:
+
+| parameter | source |
+| --------- | ------ |
+| directory | package manifest directory |
+| outputDir | default `bundle` dir or flag `--bundle-dir` |
+| packageName | packageName from `*.package.yaml` |
+| channels | ??? |
+| channelDefault | defaultChannel from `*.package.yaml` or `stable` |
+| overwrite | `--overwrite` parameter |
+
+#### Creating bundle image
+
+If we choose to build bundle images, then we can leverage the `BuildBundleImage`
+function from `opm`. This takes in two arguments:
+
+- imageTag string - tag to use for your container image
+- imageBuilder string - the container image builder i.e. `docker`, `podman`, etc
+
+This would output a new bundle image.
```

Comment:
I can see people complaining about the image build command we create from built-in data/flags. On top of these suggestions, lets build in an escape hatch:
1. `<imageBuilder> build -t <imageTag> -f <parent>/<child>/bundle.Dockerfile <parent>/<child>`
2. Take whatever the user provides to a flag `--build-cmd` and run it.